tests/tests/balances.rs

```diff
@@ -38,44 +40,41 @@ async fn balance() {
     let asset_id = AssetId::BASE;
 
     // setup config
-    let mut config = Config::local_node();
-    config.chain_conf.initial_state = Some(StateConfig {
-        height: None,
-        contracts: None,
-        coins: Some(
-            vec![
-                (owner, 50, asset_id),
-                (owner, 100, asset_id),
-                (owner, 150, asset_id),
-            ]
+    let state_config = StateConfig {
+        contracts: vec![],
+        coins: vec![
+            (owner, 50, asset_id),
+            (owner, 100, asset_id),
+            (owner, 150, asset_id),
+        ]
+        .into_iter()
+        .enumerate()
+        .map(|(idx, (owner, amount, asset_id))| CoinConfig {
+            owner,
+            amount,
+            asset_id,
+            output_index: idx as u8,
+            ..Default::default()
+        })
+        .collect(),
+        messages: vec![(owner, 60), (owner, 90)]
             .into_iter()
-            .map(|(owner, amount, asset_id)| CoinConfig {
-                tx_id: None,
-                output_index: None,
-                tx_pointer_block_height: None,
-                tx_pointer_tx_idx: None,
-                maturity: None,
-                owner,
+            .enumerate()
+            .map(|(nonce, (owner, amount))| MessageConfig {
+                sender: owner,
+                recipient: owner,
+                nonce: (nonce as u64).into(),
                 amount,
-                asset_id,
+                data: vec![],
+                da_height: DaBlockHeight::from(1usize),
             })
             .collect(),
-        ),
-        messages: Some(
-            vec![(owner, 60), (owner, 90)]
-                .into_iter()
-                .enumerate()
-                .map(|(nonce, (owner, amount))| MessageConfig {
-                    sender: owner,
-                    recipient: owner,
-                    nonce: (nonce as u64).into(),
-                    amount,
-                    data: vec![],
-                    da_height: DaBlockHeight::from(1usize),
-                })
-                .collect(),
-        ),
-    });
+        ..Default::default()
+    };
+    let config = Config {
+        state_reader: StateReader::in_memory(state_config, MAX_GROUP_SIZE),
```

Comment:
We propagate `MAX_GROUP_SIZE` in a lot of places. It would be nice if we somehow only defined it in one place and were using it everywhere. I mean, maybe `StateReader::in_memory` may don't require it

Reply:
I reckon we still want to control how large the batches are that end up in db transactions. I've made it optional for now and default to MAX on None so we avoid propagating. What do you think?

Reply:
Yeah, it works. Also we can add a separate method for tests=)

Reply:
Good point, the repeated `None` wasn't pretty either